### PR TITLE
User authentication loggin in

### PR DESCRIPTION
**Pull request by valerietrudell**
Adds logging in and out 
Adds user authentication
Fixes how flash messages are displayed

### CODE REVIEW

**Comment by valerietrudell** [comment]
Just to let you know there were merge conflicts that needed fixing (only additions no actual conflicts) and I resolved them. Branch is able to merge now

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+ class SessionsController < ApplicationController
+
+  def create
+    user = User.find_by(username: params[:session][:username])
+    if user && user.authenticate(params[:session][:password])
+      session[:user_id] = user.id
+      flash[:notice] = "Logged in as #{user.username}"
+      redirect_to dashboard_path
+    else
+      flash.now[:notice] = "Your email or password"
```
Comment (meyerhoferc) [inline app/controllers/sessions_controller.rb:10]: I really like how specific you are with the flash messages

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      render :new
+    end
+  end
+
+  def destroy
+    session.clear
+    flash[:notice] = "Logged Out"
+    redirect_to login_path   #change maybe
```
Comment (meyerhoferc) [inline app/controllers/sessions_controller.rb:18]: comment noted

File: app/views/shared/_nav.html.erb [hunk 3]
```diff
@@ -3,6 +3,12 @@
      <div class ="container">
        <a href="<%= root_path %>" class="brand-logo">Weasleys' Wizard Wheezes</a>
        <ul id="nav-mobile" class="right hide-on-med-and-down">
+          <li><a href="/cart" >Cart</a></li>
+          <% if current_user %>
+              <li><%= link_to "Logout", logout_path, method: :delete %></li>
+          <% else  %>
+            <li><%= link_to "Login", login_path, class: "login-link "%></li>
+          <% end  %>
```
Comment (meyerhoferc) [inline app/views/shared/_nav.html.erb:11]: excellent use of current_user 🔥

File: spec/features/users/user_can_login_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    user = User.create(first_name: "Charlotte",
+                       last_name: "Moore",
+                       username: "Cj",
+                       email: "email@email.com",
+                       password: "password")
+    visit login_path
+    fill_in "session[username]", with: user.username
+    fill_in "session[password]", with: user.password
+
+    save_and_open_page
```
Comment (meyerhoferc) [inline spec/features/users/user_can_login_spec.rb:15]: probably need to delete this

File: spec/features/users/user_can_login_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    within(".login-submit") do
```
Comment (meyerhoferc) [inline spec/features/users/user_can_login_spec.rb:16]: I like the css selector on the test, we probably need to make this standard for our tests

**Pull request merged**